libraries/ESP8266WiFi/src/WiFiUdp.h

```diff
@@ -47,7 +48,8 @@ class WiFiUDP : public UDP, public SList<WiFiUDP> {
   // Finish with the UDP connection
   void stop() override;
   // join a multicast group and listen on the given port
-  uint8_t beginMulticast(IPAddress interfaceAddr, IPAddress multicast, uint16_t port);
+  virtual uint8_t beginMulticast(IPAddress interfaceAddr, uint16_t port);
+  uint8_t beginMulticast(IPAddress interfaceAddr, IPAddress multicast, uint16_t port) __attribute__((deprecated));
```

Comment:
Do we need to deprecate, really, if methods do not overlap?
Virtual now works via `UDP` pointer for your case, and `WiFiUDP` user continues to use this one.
(localIP() might not be the only thing happening, any IP available to lwip stack could be used)

Reply:
Yes, ideally we could leave the two calls since they are clean overloads.
However it seems that the signature with two parameters is out there on the AVR core from 2017:

https://github.com/arduino/ArduinoCore-avr/commit/5a05bf01f4ffbe16670eb8d713688d059c3bd084

and the STM32 followed the same standard in 2019:

https://github.com/stm32duino/Arduino_Core_STM32/pull/401

However the change that introduced the new argument in the ESP8266 core was way older, back to 2015:

https://github.com/esp8266/Arduino/commit/fea362a3b53507ece698717b7f869f6befefd6b4

(and it was using `WiFi.localIP()` internally in the commit immediately before the change, not `IP_ADDR_ANY`).

I don't know, if the idea behind the Arduino core is to let developers to write the most "portable" code, I think we should push for a standard, and avoid APIs that are very specific for a specific platform only, especially at these levels of abstraction. For instance, why not having both the overloads at the base `UDP` level? 
Ideally, it would be more logic having the overload with 3 arguments declared as the virtual "abstract" one, and then the other smaller overloads directly defined at `UDP` base class as non-virtual functions using the default argument values.

So I feel that we should actually trying to converge to a single portable API between cores, making easier for library devs to write the code only once (and telling the ESP8266 devs that a more standard API is available for portability).

BTW let me remove the warning, everything could be done in a separate thread if really needed.
Thx!  L